packages/marshal/src/deeplyFulfilled.js

```diff
@@ -31,51 +68,64 @@ const { fromEntries } = Object;
  *
  * If `val` or its parts are non-key Passables only *because* they contains
  * promises, the deeply fulfilled forms of val or its parts may be keys. This
- * is for the higher "store" level of abstraction to determine, because it
- * defines the "key" notion in question.
- *
- * // TODO: That higher level is in the process of being migrated from
- * // `@agoric/store` to `@endo/patterns`. Once that is far enough along,
- * // revise the above comment to match.
- * // See https://github.com/endojs/endo/pull/1451
+ * is for the higher "@endo/patterns" level of abstraction to determine,
+ * because it defines the `Key` notion in question.
  *
- * @param {any} val
- * @returns {Promise<Passable>}
+ * @template {Passable} [T=Passable]
+ * @param {T} val
+ * @returns {Promise<DeeplyAwaited<T>>}
  */
 export const deeplyFulfilled = async val => {
+  // TODO Figure out why we need these at-expect-error directives below
+  // and fix if possible.
+  // https://github.com/endojs/endo/issues/1257 may be relevant.
+
   if (!isObject(val)) {
-    return val;
+    const prim = /** @type {Primitive} */ (val);
+    // @ts-expect-error not assignable to type 'DeeplyAwaited<T>'
+    return prim;
```

Comment:
Does it work if you just assert the maximally broad return type?
```suggestion
    return /** @type {DeeplyAwaited<T>} */ (val);
```

Reply:
Done